src/components/AddItem.js

```diff
@@ -1,11 +1,102 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { db } from '../lib/firebase';
+
+const initialFormState = {
+  itemName: '',
+  timeFrame: 7,
+  lastPurchased: null,
+};
 
 const AddItem = () => {
+  const token = 'user token';
+  const history = useHistory();
+  const [formData, setFormData] = useState(initialFormState);
+  const [error, setError] = useState('');
+  console.log(error);
+
+  const handleFormChange = (event) => {
+    event.preventDefault();
+    setFormData({
+      ...formData,
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!formData.itemName) {
+      setError(`${event.target.name} field cannot be empty`);
+    } else {
+      db.collection(token).add(formData);
+      history.push('/list');
+      setFormData(initialFormState);
+      setError('');
+    }
+  };
+
   return (
     <div>
       <h1>Add Item</h1>
+      <form onSubmit={handleSubmit}>
+        <h2>Add your item</h2>
+        <label>
+          Item Name
+          <input
+            type="text"
+            placeholder="add your item here"
+            name="itemName"
+            value={formData.itemName}
+            onChange={handleFormChange}
+          />
+          {error && <span>{error}</span>}
```

Comment:
Agree with @MikeBLambert here. I also think adding `type="submit"` to the button will make it so the user can't click the button until the required field has value (I could be wrong about that, though).

Reply:
I'm wrong about that 😂   Still a good idea to add type="submit"

Reply:
Good point @MikeBLambert. Adding `required` brings up a popup and styling around the input box. The way it's implemented now might need a little styling to take place to make the error message stand out. Do we want a default or something custom made? The current way works and I'm all for how the team wants to move forward with this.

Reply:
> One alternative to adding an error text here would be take advantage of the awesome power and might that is HTML5 and pass a `required` prop to the input field. As we're in a form and using the `onSubmit`, HTML5 will show a nice little popup when any `input` fields that are `required` are empty.
> Not sure if this will fit with the design that the team wants, but just wanted to throw that out there as an option.

I didn't know about that, I do like the little pop up and its a lot less verbose than setting another state variable
And it seems the warning we've been getting has gone away since I removed setting an error state and replaced it with the required prop.

Reply:
No I lied, the waring is back 😅

Reply:
I like the little pop up and  I learned something new as well 😄